File: app/src/main/java/dev/ragnarok/fenrir/api/adapters/BooleanAdapter.java
package dev.ragnarok.fenrir.api.adapters;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.JsonPrimitive;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import java.lang.reflect.Type;

public class BooleanAdapter extends AbsAdapter implements JsonSerializer<Boolean>, JsonDeserializer<Boolean> {

    @Override
    public JsonElement serialize(Boolean src, Type typeOfSrc, JsonSerializationContext context) {
        return new JsonPrimitive(src ? 1 : 0);
    }

    @Override
    public Boolean deserialize(JsonElement src, Type srcType,
                               JsonDeserializationContext context) throws JsonParseException {
        if (!checkPrimitive(src)) {
            return false;
        }
        JsonPrimitive prim = src.getAsJsonPrimitive();
        try {
            return prim.isBoolean() && prim.getAsBoolean() || src.getAsInt() == 1;
        } catch (Exception e) {
            return src.getAsBoolean();
        }
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/api/adapters/BooleanAdapter.kt
package dev.ragnarok.fenrir.api.adapters

import com.google.gson.*
import java.lang.reflect.Type

class BooleanAdapter : AbsAdapter(), JsonSerializer<Boolean>, JsonDeserializer<Boolean> {
    override fun serialize(
        src: Boolean,
        typeOfSrc: Type,
        context: JsonSerializationContext
    ): JsonElement {
        return JsonPrimitive(if (src) 1 else 0)
    }

    @Throws(JsonParseException::class)
    override fun deserialize(
        src: JsonElement, srcType: Type,
        context: JsonDeserializationContext
    ): Boolean {
        if (!checkPrimitive(src)) {
            return false
        }
        val prim = src.asJsonPrimitive
        return try {
            prim.isBoolean && prim.asBoolean || src.asInt == 1
        } catch (e: Exception) {
            src.asBoolean
        }
    }
}